syntax = "proto2";

package rv.protos;

import "google/protobuf/struct.proto";

message LabelStoreConfig {
    message SemanticSegmentationRasterStore {
        message BuildingOptions {
            optional float min_aspect_ratio = 1 [default=1.618];
            optional float min_area = 2;
            optional float element_width_factor = 3 [default=0.5];
            optional float element_thickness = 4 [default=0.001];
        }

        message VectorOutput {
            optional int32 denoise = 1 [default=0];
            optional string uri = 2 [default=""];
            required string mode = 3;
            required int32 class_id = 4;
            optional BuildingOptions building_options = 5;
        }

        optional string uri = 1;

        // Flag that determines whether or not the values in this
        // raster are RGB values. If so, the RGB values will be
        // translated into class IDs based on the class map.
        optional bool rgb = 2 [default=false];

        repeated VectorOutput vector_output = 3;
    }

    required string store_type = 1;
    oneof label_store_config {
        string uri = 2;
        SemanticSegmentationRasterStore semantic_segmentation_raster_store = 3;
        google.protobuf.Struct custom_config = 4;
    }
}
